### ISSUE TITLE
Zed Terminal opens bash with Raycast, but with zsh(default shell) with Spotlight.

### ISSUE THREAD

**Issue by thapabishwa-plerionaut** [zed-industries/zed#8794]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed `settings.json`

```
// Zed settings
//
// For information on how to configure Zed, see the Zed
// documentation: https://zed.dev/docs/configuring-zed
//
// To see all of Zed's default settings without changing your
// custom settings, run the `open default settings` command
// from the command palette or from `Zed` application menu.
{
  "theme": {
    "light": "Ayu Light",
    "dark": "Ayu Dark"
  },
  "vim_mode": true,
  "ui_font_size": 16,
  "buffer_font_size": 16,
  "terminal": {
    "blinking": "terminal_controlled",
    "copy_on_select": false,
    "working_directory": "current_project_directory"
  },
  "format_on_save": "on",
  "lsp": {
    "gopls": {
      "initialization_options": {
        "gofumpt": true
      }
    }
  }
}

```

Video:
https://github.com/zed-industries/zed/assets/124118626/78034b9a-25fc-45b5-a60f-69bbc8f224c3

### Environment

MacBook Pro
13-inch, M2, 2022
Chip: M2
Memory: 16 GB

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by Moshyfawn** [comment]
The default shell to open in Zed is set to "system". What happens if you change it to one of your preferred shells?

```json
{
  "terminal": {
    "shell": {
      "program": "bash"
    }
  }
}
```

**Comment by thapabishwa-plerionaut** [comment]
Hey @Moshyfawn, I see the following with both spotlight and raycast.

![image](https://github.com/zed-industries/zed/assets/124118626/fdf6ea39-4fa5-465c-adaf-b65349101d3e)

**Comment by mrnugget** [comment]
Hey there! Few questions to try to nail this down:

1. What happens if you open it by clicking on `Zed.app` in `/Applications`?
2. And what happens if you launch it via the CLI (if you have the CLI tools installed)? 
3. What do you have set as your login shell in the macOS User settings? (You need to right click and "Advanced options..." to see these)
    ![screenshot-2024-03-04-10 37 35@2x](https://github.com/zed-industries/zed/assets/1185253/49a7905e-73d6-46dd-83b4-cf0aaab1eed6)
4. What shell do you get in `Terminal.app`?

**Comment by thapabishwa-plerionaut** [comment]
> What happens if you open it by clicking on Zed.app in /Applications?

Works as expected.

> And what happens if you launch it via the CLI (if you have the CLI tools installed)?

Works as expected via CLI too.

> What do you have set as your login shell in the macOS User settings? (You need to right click and "Advanced options..." to see these)


<img width="827" alt="image" src="https://github.com/zed-industries/zed/assets/124118626/6cca1fa3-e254-4b22-8824-6d76e95fa36e">


> What shell do you get in Terminal.app?

Terminal.app
![image](https://github.com/zed-industries/zed/assets/124118626/62d93434-7139-47f0-822b-621dda6cca52)

iterm2.app
![image](https://github.com/zed-industries/zed/assets/124118626/06285142-1b73-4069-b87d-4393fe249ba4)

**Comment by mrnugget** [comment]
Huh. So what happens when you start [Alacritty.app](https://github.com/alacritty/alacritty)? (That's what we use for the built-in terminal)

Do you have any plugins in Raycast that set a `$SHELL` or something?

**Comment by aspeth** [comment]
+1 to this issue - when opened via Raycast, Zed says my shell is `/bin/sh` despite having it set to `/bin/zsh` in my settings.json, but when opened via Finder or Spotlight it respects the setting. My login shell in the macOS User settings is `/bin/zsh` and I don't believe I have any Raycast plugins that set a `$SHELL`

Not a huge problem since things work as expected when opened with anything that's not Raycast, just wanted to add my experience here.

**Comment by mrnugget** [comment]
@aspeth can you try to reproduce what happens when you just launch Alacritty.app, from the Dock and from Raycast?

**Comment by aspeth** [comment]
`echo $SHELL` in Alacritty shows `/bin/zsh` when opened via Raycast as well as from the dock, but when opened via Raycast can't find commands in my `.zshrc` (brew, starship, and atuin, specifically) even after running `source ~/.zshrc` but has no problem finding those commands when opened via the dock.

**Comment by mrnugget** [comment]
This is so weird. I really wonder whether there isn't something going on inside Raycast wrt to env variables.

**Comment by tkgalk** [comment]
I'm having a similar issue. Zed is trying to open `fish` and fails because fish is no longer installed on the system. My shell is set to "system", my default shell is zsh. Nothing in the Zed's settings seems to be pointing to fish, but it keeps trying to use it.

**Comment by mrnugget** [comment]
> I'm having a similar issue.

When you open it via Raycast?

**Comment by tkgalk** [comment]
> When you open it via Raycast?

No. When running Zed in general. It is a different case, I don't even have Raycast, but Zed trying to pick a weird shell is similar. I thought I'd mention it just on the off chance it helps with debugging.

**Comment by mrnugget** [comment]
@tkgalk can you also try running [Alacritty.app](https://github.com/alacritty/alacritty)? I'm wondering if you have the same problem there. That would help a lot with the debugging.

**Comment by tkgalk** [comment]
https://github.com/zed-industries/zed/discussions/11751 moved my issue here, as it doesn't seem to be connected to Raycast.

**Comment by tkgalk** [comment]
After fixing my issue with Zed, I installed Raycast just to help, but I can't reproduce the OP's problem. Zed seems to be correctly running the shell from Raycast.

1. Configured to shell: system.
2. `/bin/zsh`.
3. `/bin/zsh`.
4. `/bin/zsh`.
5. (It's `unset SHELL`, I think, not `$SHELL`). `name: tkg dir: /Users/tkg shell: /bin/zsh`
6. `/bin/zsh`.

So at least for me seems like Zed is behaving correctly through both Raycast and outside of it.

**Comment by mrnugget** [comment]
Figured out a little bit more my digging into Alacritty and compiling from source:

1. When you `chsh -s /new/-shell`
2. Start Alacritty from Finder

That new shell is not picked up, because `Finder` _does_ set the `$SHELL` env variable, but it only gets updated once Finder is restarted.

There's so much environment variables and state involved here, especially when using Raycast/Finder (which are long-running applications) that I think the safest step for further reproduction of bugs is: *Make sure you've restarted your computer after making changes to the default shell.*
